.github/actions/utility-action/action.yml

```diff
@@ -0,0 +1,112 @@
+name: Utility Actions
+
+description: |
+  A set of utility steps to be used across different workflows, including:
+  - Installing Terraform
+  - Installing AWS CLI
+  - Setting Terraform variables
+  - Checking/Creating an S3 bucket
+
+inputs:
+  awscli-version:
+    description: 'Version of the AWS CLI to install'
+    required: true
+    # renovate: datasource=github-releases depName=aws/aws-cli
+    default: '2.15.52'
+  terraform-version:
+    description: 'Version of Terraform to install'
+    required: true
+    default: 'latest'
+  s3-backend-bucket:
+    description: 'Name of the S3 bucket to store Terraform state'
+    required: true
+  s3-bucket-region:
+    description: 'Region of the bucket containing the resources states, if not set, will fallback on aws-region'
+  aws-region:
+    description: 'AWS region to use for S3 bucket operations'
+    required: true
+  name:
+    description: 'Name of resource instance (e.g., uid)'
```

Comment:
Maybe we could add here something along the lines of.

`... Used as unique value for the TF state path in S3.`

I don't think the `name` has any other functional use.